openapi_python_client/templates/endpoint_macros.pyi

```diff
@@ -45,25 +45,28 @@ if {% if not property.required %}not isinstance({{ property_name }}, Unset) and
 {% endif %}
 {% endmacro %}
 
+{% macro encode_body(property, destination) %}
+{% if property.template %}
+{% import "property_templates/" + property.template as prop_template %}
+{% if prop_template.transform_extended %}
```

Comment:
This is the same kind of vaguely-polymorphic use of Jinja that I've done in some other PRs. The `transform_extended` macro, which has the extra parameter for skipping read-only properties, only exists in `model_property.pyi`. If we're using a different template that doesn't have that, then it'll go to line 54 instead.